.ci/all.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Attention, there is no "-x" to avoid problems on Travis
+set -e
+
+# result of command goes to $RUN_CI
+# 0 = skip
+# 1 = no skip
+function should_run_ci {
+    SKIP_CI_BY_FILES=$1
+    SKIP_FILES=$2
+    DEBUG=$3
+
+    echo "Initial value of SKIP_CI_BY_FILES: $SKIP_CI_BY_FILES"
+    echo "Files to that are allowed to be skipped: $SKIP_FILES"
+
+    if [[ $SKIP_CI_BY_FILES != 'false' ]]; then
+         ## Identify previous commit to know how much to check
+         ## Check assumes we are only working with 1 commit if we are in master
+         PREVIOUS_COMMIT="HEAD~1"
+
+         if [[ $DEBUG == "true" ]]; then
+              OUT=`git rev-parse HEAD`
+              echo "Head commit: $OUT"
+         fi
+         if [[ $DEBUG == "true" ]]; then
+              OUT=$(git branch master --contains `git rev-parse HEAD`)
+              echo "Master contains head commit: $OUT"
+         fi
+
+         if [[ $(git branch master --contains `git rev-parse HEAD` \
+                    | wc -c ) == 0 ]]; then
+              PREVIOUS_COMMIT=`git merge-base origin/master HEAD`
+         fi
+
+         echo "Previous Commit to start with: $PREVIOUS_COMMIT"
+
+         if [[ $DEBUG == "true" ]]; then
+              OUT=`git diff --name-only HEAD $PREVIOUS_COMMIT`
+              echo "Files between top commit and previous: $OUT"
+         fi
+         if [[ $DEBUG == "true" ]]; then
+              OUT=`git diff --name-only HEAD $PREVIOUS_COMMIT | grep -vE "$SKIP_FILES"`
+              echo "Files should not skip: $OUT"
+         fi
+
+         if [[ $(git diff --name-only HEAD $PREVIOUS_COMMIT \
+                    | grep -vE "$SKIP_FILES" | cat | wc -c | sed 's/^ *//' ) > 0 ]]; then
+              SKIP_CI_BY_FILES="false"
+         else
+              SKIP_CI_BY_FILES="true"
+         fi
+    else
+        echo "CI step logic doesn't want a skip by files"
+    fi
+
+    echo "SKIP_CI_BY_FILES: $SKIP_CI_BY_FILES"
+
+    SKIP_CI_BY_COMMIT="false"
+
+    if [[ $DEBUG == "true" ]]; then
+         OUT=`git log -1 --format=%B`
+         echo "Top commit message: $OUT"
+    fi
+
+    if [ $(git log -1 --format=%B | grep -E "\[maven-release-plugin\] prepare release" \
+              | cat | wc -l) -lt 1 ];
+    then
+        SKIP_CI_BY_COMMIT="false"
+    else
+        SKIP_CI_BY_COMMIT="true"
+    fi
+
+    echo "SKIP_CI_BY_COMMIT: $SKIP_CI_BY_COMMIT"
+
+    if [[ $SKIP_CI_BY_FILES == 'false' && $SKIP_CI_BY_COMMIT == 'false' ]]; then
+         echo "CI should not be skipped."
```

Comment:
"CI should run"
to avoid word "skip" at all

Reply:
Done.